src/Core/Domain/OrderReturn/Exception/OrderReturnNotFoundException.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception;
+
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\ValueObject\OrderReturnId;
+
+/**
+ * Is thrown when order return is not found in order return subdomain
+ */
+class OrderReturnNotFoundException extends OrderReturnException
+{
+    /**
+     * @var OrderReturnId
+     */
+    private $orderReturnId;
```

Comment:
I think exception having an id of what it did not find is a good idea, because if exception happens when multiple objects are handled usually the first question is "Which ID was not found?"

But you are right I don't use it here, so I will remove it for now. I will take a look at abstract repository for a way to include id.